x-pack/metricbeat/module/aws/sqs/sqs_integration_test.go

```diff
@@ -20,7 +21,7 @@ func TestFetch(t *testing.T) {
 	config := mtest.GetConfigForTest(t, "sqs", "300s")
 
 	metricSet := mbtest.NewReportingMetricSetV2Error(t, config)
-	events, errs := mbtest.ReportingFetchV2Error(metricSet)
+	events, errs := mbtest.PeriodicReportingFetchV2Error(metricSet, 1*time.Minute, 8*time.Minute)
```

Comment:
Are we planning on replacing ReportingfetchV2Error with PeriodicReportingFetchV2Error in other metricsets?

Reply:
I want to focus this PR on removing the flakiness from the test suite so that other teams can continue with their tasks. 

Do you know of other tests that need more than one call to `Fetch()` to succeed due to latency or something similar?

Reply:
But if this solution proves effective, we can adopt it for more tests that need more than one fetch.